// semantic-ui-vue/dist/commonjs/elements/Segment/Segment.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiSegment',
  mixins: [_lib.SemanticUIVueMixin],
  props: {
    attached: _PropTypes.Enum.Attached({
      type: Boolean,
      description: 'Attach segment to other content, like a header.'
    }),
    basic: Boolean,
    clearing: {
      type: Boolean,
      description: 'A segment can clear floated content.'
    },
    inverted: Boolean,
    padded: _PropTypes.Enum.Padded({
      type: Boolean,
      description: 'A segment can increase its padding'
    }),
    piled: Boolean,
    raised: Boolean,
    stacked: (0, _PropTypes.Enum)(['tall'], {
      type: Boolean
    }),
    vertical: Boolean,
    disabled: Boolean,
    loading: Boolean,
    compact: Boolean,
    color: _PropTypes.Enum.Color({
      type: String
    }),
    emphasis: _PropTypes.Enum.Emphasis({
      type: String
    }),
    circular: Boolean,
    floated: _PropTypes.Enum.Floated({
      type: String
    }),
    aligned: _PropTypes.Enum.TextAlign({
      type: String
    })
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = this.getElementType();
    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([this.getChildPropsAndListeners(), {
        'class': this.classes('ui', this.attached, this.attached && 'attached', this.basic && 'basic', this.clearing && 'clearing', this.padded, this.padded && 'padded', this.inverted && 'inverted', this.stacked, this.stacked && 'stacked', this.piled && 'piled', this.raised && 'raised', this.vertical && 'vertical', this.disabled && 'disabled', this.loading && 'loading', this.compact && 'compact', this.color, this.emphasis, this.circular && 'circular', this.floated, this.floated && 'floated', this.aligned, this.aligned && 'aligned', 'segment')
      }]),
      [this.$slots.default]
    );
  }
};
