// semantic-ui-vue/dist/commonjs/collections/Table/TableRow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiTableRow',
  mixins: [_lib.SemanticUIVueMixin],
  props: {
    active: {
      description: 'A row can be active or selected by a user.',
      type: Boolean
    },
    disabled: {
      description: 'A row can be disabled.',
      type: Boolean
    },
    error: {
      description: 'A row may call attention to an error or a negative value.',
      type: Boolean
    },
    negative: {
      description: 'A row may let a user know whether a value is bad.',
      type: Boolean
    },
    positive: {
      description: 'A row may let a user know whether a value is good.',
      type: Boolean
    },
    selected: {
      description: 'DEPRECATED A row can be active or selected by a user.',
      type: Boolean
    },
    state: _PropTypes.Enum.State({
      description: 'DEPRECATED'
    }),
    textAlign: (0, _PropTypes.Enum)(['left', 'right', 'center'], {
      description: 'A table row can adjust its text alignment.'
    }),
    verticalAlign: _PropTypes.Enum.VerticalAlign({
      description: 'A row may warn a user.'
    }),
    warning: {
      description: 'A row may warn a user.',
      type: Boolean
    }
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = this.getElementType('tr');
    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([this.getChildPropsAndListeners(), {
        'class': this.classes(this.textAlign && this.textAlign + ' aligned', this.verticalAlign && this.verticalAlign + ' aligned', this.disabled && 'disabled', this.error && 'error', this.negative && 'negative', this.positive && 'positive', (this.selected || this.active) && 'active', this.warning && 'warning', this.state) }]),
      [this.$slots.default]
    );
  },

  meta: {
    parent: 'SuiTable'
  }
};
