// semantic-ui-vue/dist/commonjs/elements/Header/Header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiHeader',
  mixins: [_lib.SemanticUIVueMixin],
  props: {
    color: _PropTypes.Enum.Color(),
    content: String,
    dividing: Boolean,
    disabled: Boolean,
    floated: (0, _PropTypes.Enum)(['left', 'right']),
    icon: {
      type: [Boolean, String],
      default: false
    },
    image: {
      type: String,
      description: 'Add an image by img src or pass an Image.'
    },
    size: _PropTypes.Enum.Size(),
    sub: Boolean,
    textAlign: (0, _PropTypes.Enum)(['left', 'right', 'center', 'justify']),
    block: Boolean,
    attached: (0, _PropTypes.Enum)(['top', 'bottom'], {
      type: Boolean
    }),
    inverted: Boolean
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = this.getElementType();

    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([this.getChildPropsAndListeners(), {
        'class': this.classes('ui', this.floated && this.floated + ' floated', (0, _lib.textAlign)(this.textAlign), this.attached, this.attached && 'attached', this.color, this.size, this.icon && 'icon', this.block && 'block', this.dividing && 'dividing', this.image && 'image', this.sub && 'sub', this.disabled && 'disabled', this.inverted && 'inverted', 'header')
      }]),
      [this.icon !== !!this.icon && h(
        'sui-icon',
        {
          attrs: { name: this.icon }
        },
        []
      ), this.image && h(
        'img',
        {
          attrs: { src: this.image },
          'class': 'ui image' },
        []
      ), this.$slots.default || this.content]
    );
  }
};
